backend/src/main/java/gov/cdc/usds/simplereport/api/model/facets/LocatedWrapper.java

```diff
@@ -0,0 +1,52 @@
+package gov.cdc.usds.simplereport.api.model.facets;
+
+import gov.cdc.usds.simplereport.db.model.LocatedEntity;
+import gov.cdc.usds.simplereport.db.model.auxiliary.StreetAddress;
+import gov.cdc.usds.simplereport.service.model.Wrapper;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Mix-in interface to handle retrieving individual address fields, or the entire address object, as
+ * required.
+ *
+ * @param <T> an entity that has a street address
+ */
+public interface LocatedWrapper<T extends LocatedEntity> extends Wrapper<T> {
+
+  default StreetAddress getAddress() {
+    return getWrapped().getAddress();
+  }
+
+  default String getAddressField(Function<? super StreetAddress, String> mapper) {
```

Comment:
Love this

Reply:
It might be a bit much, honestly, but it was more fun than the alternative. 😄